src/praisonai-code/praisonai_code/cli/features/code_intelligence.py

```
@@ -579,6 +561,84 @@ async def _handle_search_code(self, query: str, file_path: str = None) -> CodeQu
+                results.append({
+                    "file": parts[0],
+                    "line": line_no,
+                    "content": parts[2] if len(parts) > 2 else ""
+                })
+        return results
+
+    def _fallback_grep(self, search_term: str, search_path: str) -> List[Dict[str, Any]]:
+        """Last-resort search using a plain ``grep -rn`` subprocess."""
+        import subprocess
+
+        result = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)